### ISSUE TITLE
Appropriately handle `lat` and `long` columns and add flexibility

### ISSUE THREAD

**Issue by njtierney** [njtierney/maxcovr#63]
At the moment `maxcovr` assumes that all columns with latitude and longitude information are named `lat` and `long`.

This is not ideal, as it throws a cryptic error message when these are not given the appropriate names. 

There should be three steps in refactoring:

1. Add a `stop` function that stops the analysis if things are not named `lat` and `long`. This can then be used in each function that uses that info.

2. Allow the user to specify their own `lat` and `long` arguments

3. Cleverly detect different variations on `lat` and `long` as `leaflet` does and display a nice message

**Comment by mpadge** [comment]
I've got this kind of thing in `dodgr` with an internal [`dodgr_graph_cols` function](https://github.com/ATFutures/dodgr/blob/master/R/graph-functions.R), which relies on a whole bunch of helper functions [here](https://github.com/ATFutures/dodgr/blob/master/R/graph-functions-misc.R). It's kinda messy, but works really well at auto-identifying standard columns based on unknown and variable input names. The idea is to return a list so everything else then works like this:
``` r
cols <- dodgr_graph_cols (graph)
lon <- my_input [[cols$lon]]
```
... or whatever. Point is, it's robust and flexible: Write a function to auto-identify the columns you need; return those as list items; then use code like the above to extract when needed.

**Comment by njtierney** [comment]
OK that is rather neat! 

Am I correct in that we are looking for something like this to identify say x and y cols:

* https://github.com/ATFutures/dodgr/blob/master/R/graph-functions-misc.R#L84 and
* https://github.com/ATFutures/dodgr/blob/master/R/graph-functions.R#L43

**Comment by mpadge** [comment]
yeah, the details are all pretty easy. The difficult and important thing was figuring out that the `get_columns` function should return a **list**. From that point on, it's all easy
